File: portal_berita_flutter/lib/models/article.dart
import 'category.dart';
import 'package:intl/intl.dart';

class Article {
  final int id;
  final String title;
  final String slug;
  final String excerpt;
  final String content;
  final String? image;
  final String author;
  final int views;
  final bool isTrending;
  final bool isFeatured;
  final DateTime publishedAt;
  final Category category;

  Article({
    required this.id,
    required this.title,
    required this.slug,
    required this.excerpt,
    required this.content,
    this.image,
    required this.author,
    required this.views,
    required this.isTrending,
    required this.isFeatured,
    required this.publishedAt,
    required this.category,
  });

  factory Article.fromJson(Map<String, dynamic> json) {
    return Article(
      id: json['id'],
      title: json['title'],
      slug: json['slug'],
      excerpt: json['excerpt'],
      content: json['content'],
      image: json['image'],
      author: json['author'],
      views: json['views'],
      isTrending: json['is_trending'] ?? false,
      isFeatured: json['is_featured'] ?? false,
      publishedAt: DateTime.parse(json['published_at']),
      category: Category.fromJson(json['category']),
    );
  }

  String get timeAgo {
    final now = DateTime.now();
    final difference = now.difference(publishedAt);

    if (difference.inMinutes < 60) {
      return '${difference.inMinutes} menit lalu';
    } else if (difference.inHours < 24) {
      return '${difference.inHours} jam lalu';
    } else {
      return '${difference.inDays} hari lalu';
    }
  }

  String get formattedDate {
    return DateFormat('d MMM yyyy, HH:mm', 'id_ID').format(publishedAt);
  }
}

File: projeklaravel1/app/Models/Article.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Article extends Model
{
    use HasFactory;

    protected $fillable = [
        'category_id',
        'title',
        'slug',
        'excerpt',
        'content',
        'image',
        'author',
        'views',
        'is_trending',
        'is_featured',
        'published_at'
    ];

    protected $casts = [
        'published_at' => 'datetime',
        'is_trending' => 'boolean',
        'is_featured' => 'boolean',
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function comments()
    {
        return $this->hasMany(Comment::class)->where('is_approved', true)->latest();
    }

    public function getTimeAgoAttribute()
    {
        $diff = $this->published_at->diffInMinutes(now());
        
        if ($diff < 60) {
            return $diff . ' menit lalu';
        } elseif ($diff < 1440) {
            return floor($diff / 60) . ' jam lalu';
        } else {
            return floor($diff / 1440) . ' hari lalu';
        }
    }

    public function incrementViews()
    {
        $this->increment('views');
    }
}
